src/extension/authkey/src/test/java/org/geoserver/security/AuthKeyAuthenticationTest.java

```diff
@@ -655,6 +655,38 @@ public void testAuthKeyMapperSynchronize() throws Exception {
         assertNull(userpropMapper.getUser(user2KeyB));
     }
 
+    @Test
+    public void testAuthKeyMapperAutoSynchronize() throws Exception {
+        String authKeyUrlParam = "myAuthKey";
+        String filterName = "testAuthKeyFilterAuto1";
+
+        AuthenticationKeyFilterConfig config = new AuthenticationKeyFilterConfig();
+        config.setClassName(GeoServerAuthenticationKeyFilter.class.getName());
+        config.setName(filterName);
+        config.setAllowMapperKeysAutoSync(true);
+        config.setUserGroupServiceName("ug1");
+        config.setAuthKeyParamName(authKeyUrlParam);
+        config.setAuthKeyMapperName("propertyMapper");
+
+        getSecurityManager().saveFilter(config);
+
+        // File Property Mapper
+        File authKeyFile = new File(getSecurityManager().userGroup().dir(), "testAuthKey");
+        authKeyFile = new File(authKeyFile, "authkeys.properties");
+
+        GeoServerAuthenticationKeyProvider geoServerAuthenticationKeyProvider =
+                new GeoServerAuthenticationKeyProvider(getSecurityManager(), 10);
+        assertNotNull(geoServerAuthenticationKeyProvider.getScheduler());
+        assertFalse(geoServerAuthenticationKeyProvider.getScheduler().isTerminated());
+        assertFalse(geoServerAuthenticationKeyProvider.getScheduler().isShutdown());
+        assertEquals(10, geoServerAuthenticationKeyProvider.getAutoSyncDelaySeconds());
+        Thread.sleep(2 * geoServerAuthenticationKeyProvider.getAutoSyncDelaySeconds() * 1000L);
```

Comment:
Due to runtime issues and Github actions VM unpredictability, waiting exactly the time configured might not be enough. Check the code for usage of Awaitility, it helps set up a wait for a given condition, and with a max timeout that allows for some tolerance to slow running environments.

Reply:
Done, thanks for the hint.